src/enigma/reload-in-progress-interceptor.js

```diff
@@ -0,0 +1,87 @@
+import { useState, useEffect } from 'react';
+
+const ERR_RELOAD_IN_PROGRESS = 11000;
+let wasInReloadOnStartup;
+let reloadInProgress;
+let setReloadInProgress;
+
+function sleep(time) {
+  return new Promise(resolve => setTimeout(resolve, time));
+}
+function useReloadInProgress(app) {
+  [reloadInProgress, setReloadInProgress] = useState(wasInReloadOnStartup);
+  useEffect(() => {
+    if (!app) return null;
+
+    const modelChanged = async () => {
+      try {
+        await app.getAppLayout();
+        if (reloadInProgress) {
+          setReloadInProgress(false);
+        }
+        // When a change is triggered by a reload the getAppLayout sometimes resolve quick enough to not throw a reload in progress error, so call it
+        // once more to increase the likelyhood a bit. If this too fails any interaction will immediately pop up the reload in progress dialog
+        await sleep(1000);
+        await app.getAppLayout();
+      } catch (err) {
+        if (err.code === ERR_RELOAD_IN_PROGRESS) {
+          if (!reloadInProgress) {
+            setReloadInProgress(true);
+          }
+        }
+      }
+    };
+
+    app.on('changed', modelChanged);
+    modelChanged();
+
+    return () => {
+      app.removeListener('changed', modelChanged);
+    };
+  }, [app && app.id]);
+
+  return reloadInProgress;
+}
+
+function retryUsingTimeouts(request) {
+  return new Promise(((resolve/* , reject */) => {
+    async function retry() {
+      try {
+        const res = await request.retry();
+        resolve(res);
+      } catch (err) {
+        if (err.code === ERR_RELOAD_IN_PROGRESS) {
+          setTimeout(() => {
+            retry();
+          }, 500);
+        }
+      }
+    }
+
+    setTimeout(() => {
+      retry();
+    }, 500);
+  }));
+}
+const interceptor = {
+  onRejected(session, request, error) {
+    if (error.code === ERR_RELOAD_IN_PROGRESS) {
+      if (!reloadInProgress) {
+        if (setReloadInProgress) {
+          setReloadInProgress(true);
+        } else {
+          wasInReloadOnStartup = true;
```

Comment:
Wouldn't `wasInReloadOnStartup` only be `true` if it's the `GetActiveDoc` or `OpenDoc` methods?